test/testautovariables.cpp

```diff
@@ -775,6 +776,19 @@ class TestAutoVariables : public TestFixture {
         ASSERT_EQUALS("[test.cpp:4]: (error) Address of auto-variable 'x' returned\n", errout.str());
     }
 
+    void returnLocalVariable7() {
+        check("struct s {\n"
+              "    int f;\n"
+              "};\n"
+              "int *foo(int *v) {\n"
+              "    struct s *p = malloc(sizeof *p);\n"
+              "    p->f = *v;\n"
```

Comment:
it's not necessary to assign `p->f` as far as I see.

Reply:
agree, but wanted to fool proof the test case against some possible (value not initialized) concern in the future.

same goes with the malloc, but IMHO helps visualize why this variable couldn't be considered "auto-variable" and so why the valueflow (I suspect) logic is buggy.

apologize for not being able to come up yet with a fix otherwise